app/calendar/event-creator.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+import { useState } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+export default function EventCreator({ username, onEventCreated }) {
+  const [name, setName] = useState("");
+  const [date, setDate] = useState("");
+  const [calendarId, setCalendarId] = useState("");
+
+  function validateInput() {
+    var valid = true;
+    var errstyle = "2px solid red";
+    var initstyle = "1px solid grey";
+
+    if (name.length == 0) {
```

Comment:
```suggestion
    if (name.length === 0) {
```
See [this](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons) stackoverflow answer for more details.